### PR TITLE
Added new RCF `bucket` for GCS and ABS

### PR DESCRIPTION
**Pull request by akanshi-elastic**
#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

### CODE REVIEW

File: connectors/sources/azure_blob_storage.py [hunk 1]
```diff
@@ -73,11 +74,17 @@ def get_default_configuration(cls):
                 "order": 3,
                 "type": "str",
             },
+            "container": {
+                "display": "textarea",
+                "label": "Azure Blob Storage container",
```
Comment (timgrein) [inline connectors/sources/azure_blob_storage.py:79]: Should this be container**s**, if the `type` is `list`?

File: connectors/sources/google_cloud_storage.py [hunk 2]
```diff
@@ -172,11 +172,17 @@ def get_default_configuration(cls):
         """
 
         return {
+            "bucket": {
+                "display": "textarea",
+                "label": "Google Cloud Storage bucket",
```
Comment (timgrein) [inline connectors/sources/google_cloud_storage.py:177]: Same as above: Should this be "[...] bucket**s**"?

File: connectors/sources/azure_blob_storage.py [hunk 3]
```diff
@@ -237,12 +244,43 @@ async def get_blob(self, container):
...
             dictionary: Documents from Azure Blob Storage
         """
-        async for container in self.get_container():
-            async for blob in self.get_blob(container=container):
-                yield blob, partial(self.get_content, blob)
+        if self.containers == ["*"]:
+            async for container_data in self.get_container():
+                async for blob in self.get_blob(container=container_data):
+                    yield blob, partial(self.get_content, blob)
+        else:
```
Comment (wangch079) [inline connectors/sources/azure_blob_storage.py:279]: I think it's much more efficient to enable filtering of `get_container`:
```python
async def get_container(self, container_list):
    container_set = set(container_list)
    async with BlobServiceClient.from_connection_string(
        conn_str=self.connection_string, retry_total=self.retry_count
    ) as azure_base_client:
        async for container in azure_base_client.list_containers(
            include_metadata=True
        ):
            if '*' in container_set or container["name"] in container_set:
                yield {
                    "name": container["name"],
                    "metadata": container["metadata"],
                }

            if '*' not in container_set and container["name"] in container_set:
                container_set.remove(container["name"])
                if not container_set:
                    return
```

Comment (akanshi-elastic) [reply connectors/sources/azure_blob_storage.py:279]: @wangch079 FYI , with this logic we won't be able to log for for invalid container.

Comment (wangch079) [reply connectors/sources/azure_blob_storage.py]: You can log at the end if `container_set` is not empty.

File: connectors/sources/azure_blob_storage.py [hunk 4]
```diff
@@ -196,22 +203,37 @@ async def blob_download_func(self, blob_name, container_name):
...
+                ):
+                    if "*" in container_set or container["name"] in container_set:
+                        yield {
+                            "name": container["name"],
+                            "metadata": container["metadata"],
+                        }
+                    if "*" not in container_set and container["name"] in container_set:
+                        container_set.remove(container["name"])
+                        if not container_set:
+                            yield
```
Comment (wangch079) [inline connectors/sources/azure_blob_storage.py:230]:
```suggestion
                            return
```

File: connectors/sources/azure_blob_storage.py [hunk 5]
```diff
@@ -196,22 +203,37 @@ async def blob_download_func(self, blob_name, container_name):
...
+                if container_set:
+                    self._logger.warning(f"{container_set} is invalid. Skipping.")
```
Comment (wangch079) [inline connectors/sources/azure_blob_storage.py:232]: You should log something like `Container(s) x, y, z are configured but not found`

File: connectors/sources/google_cloud_storage.py [hunk 6]
```diff
@@ -378,11 +384,19 @@ async def get_docs(self, filtering=None):
...
+                ):
+                    for blob_document in self.get_blob_document(blobs=blobs):
+                        yield blob_document, partial(self.get_content, blob_document)
+        else:
+            for bucket in self.configuration["buckets"]:
+                async for blobs in self.fetch_blobs(
+                    buckets={"items": [{"id": bucket, "name": bucket}]},
+                ):
+                    for blob_document in self.get_blob_document(blobs=blobs):
+                        yield blob_document, partial(self.get_content, blob_document)
```
Comment (wangch079) [inline connectors/sources/google_cloud_storage.py:402]: One more suggestion, you can make a change to `fetch_buckets()`:

```python
async def fetch_buckets(self, buckets):
    if "*" in buckets:
        async for bucket in self._google_storage_client.api_call(
            resource="buckets",
            method="list",
            full_response=True,
            project=self._google_storage_client.user_project_id,
            userProject=self._google_storage_client.user_project_id,
        ):
            if bucket.get("items")
                yield bucket
    else:
        for bucket in buckets:
            yield {"items": [{"id": bucket, "name": bucket}]}
```

Then you have simplify this method:
```suggestion
        async for bucket in self.fetch_buckets(self.configuration["buckets"]):
            async for blobs in self.fetch_blobs(buckets=bucket):
                for blob_document in self.get_blob_document(blobs=blobs):
                    yield blob_document, partial(self.get_content, blob_document)
```

**Review by wangch079 — APPROVED** [review]

**Pull request merged**